config/database.yml

```diff
@@ -1,14 +1,13 @@
-development:
+development: &default
   adapter: postgresql
   encoding: unicode
   database: planner_development
   pool: 5
+  username: postgres
+  password:
+  host: db
```

Comment:
I would prefer this to be localhost so we can run the app with or without docker. Any ideas?

Reply:
Even if you forward the PostgreSQL port, `localhost` still won't work because the Rails server is running in its own container with its own idea of what _localhost_ is. So I suggest an environment variable called `DB_HOST` which is used here, with a fallback of `localhost`.

No idea how to do that in a YAML file though. Perhaps the whole config needs to be overridden? If you set `RAILS_ENV` to `development-docker` in _docker-compose.yml_, and override here:

```yaml
development-docker:
  <<: *default
  host: db
```

Will that work?

Violation of [Rule III](https://12factor.net/config) makes me sad.

Reply:
Looks like [you can embed ERB in _config/database.yml_](http://guides.rubyonrails.org/configuring.html#connection-preference) so this might work:

```yaml
   host: <%= JSON.generate(ENV['DB_HOST'] || 'localhost') %>
```

Reply:
https://github.com/despo/planner/pull/4